q2_moshpit/eggnog/_dbs.py

```diff
@@ -113,28 +113,205 @@ def fetch_diamond_db() -> DiamondDatabaseDirFmt:
+            "http://eggnog5.embl.de/download/eggnog_5.0/e5.taxid_info.tsv"
+        ]
+    )
+
+    # Let user know that the process is done.
+    # The actual copying will be taken care of by qiime behind the
     # scenes.
-    print(
-        colorify(
-            "Decompression completed. \n"
-            "Copying file from temporary directory to final location "
-            "(this will take a few minutes)...",
-            "lgreen"
+    print(colorify(
+        "Download completed. \n"
+        "Copying files from temporary directory to final location "
+        "(this will take a few minutes)..."
+    ))
+
+    return eggnog_fa
+
+
+def build_eggnog_diamond_db(
+        eggnog_proteins: EggnogProteinSequencesDirFmt,
+        taxon: int
+) -> DiamondDatabaseDirFmt:
+    """
+    Creates an DIAMOND database which contains the protein
+    sequences that belong to the specified taxon.
+    """
+    # Validate taxon ID
+    _validate_taxon_id(eggnog_proteins, taxon)
+
+    # Initialize output objects
+    diamond_db = DiamondDatabaseDirFmt()
+
+    # Define command.
+    cmd = [
+        "create_dbs.py",
+        "--data_dir", str(eggnog_proteins),
+        "--taxids", str(taxon),
+        "--dbname", "ref_db"
+    ]
+    run_command(cmd)
+
+    # The script will create the diamond DB in side the directory of
+    # eggnog_proteins object, so we need to move it to diamond_db
+    source_path = os.path.join(str(eggnog_proteins), "ref_db.dmnd")
+    destination_path = os.path.join(str(diamond_db), "ref_db.dmnd")
+    shutil.move(source_path, destination_path)
+
+    # Return objects
+    return diamond_db
+
+
+def _validate_taxon_id(eggnog_proteins, taxon):
+    # Validate taxon id number
+    # Read in valid taxon ids
+    taxid_info = pd.read_csv(
+        os.path.join(str(eggnog_proteins), "e5.taxid_info.tsv"),
+        sep="\t"
+    )
+
+    # Convert them into a set
+    tax_ids = set()
+    for lineage in taxid_info["Taxid Lineage"]:
+        tax_ids.update(
+            set(
+                lineage.strip().split(",")
+            )
+        )
+
+    # Check for overlap with provided taxon id
+    if not tax_ids.intersection(set(str(taxon))):
+        raise ValueError(
+            f"'{taxon}' is not valid taxon ID. "
+            "To view all valid taxon IDs inspect e5.taxid_info.tsv "
+            "file in the input eggnog_proteins input."
         )
+
+
+def fetch_ncbi_taxonomy() -> NCBITaxonomyDirFmt:
+    """
+    Script fetches 3 files from the internet and puts them into the folder of
+    a NCBITaxonomyDirFmt object.
+    """
+    # Initialize output object and paths
+    ncbi_data = NCBITaxonomyDirFmt()
+    zip_path = os.path.join(str(ncbi_data), "taxdmp.zip")
+    nodes_path = os.path.join(str(ncbi_data), "nodes.dmp")
+    names_path = os.path.join(str(ncbi_data), "names.dmp")
+    proteins_path = os.path.join(str(ncbi_data), "prot.accession2taxid.gz")
+    version_path = os.path.join(str(ncbi_data), "version.tsv")
```

Comment:
```suggestion
file_names = ["taxdmp.zip", "nodes.dmp", "names.dmp", "prot.accession2taxid.gz", "version.tsv"]
file_paths = [os.path.join(str(ncbi_data), file_name) for file_name in file_names]
zip_path, nodes_path, names_path, proteins_path, version_path = file_paths
```
You could add a loop here, to not repeat yourself.

Reply:
@VinzentRisch yeah... also thought about it but... let's brake it down. 

Option 1 (your proposal adjusted to fit in 79 columns):
```python
file_names = [
    "taxdmp.zip", "nodes.dmp", "names.dmp",
    "prot.accession2taxid.gz", "version.tsv"
]
file_paths = [
    os.path.join(str(ncbi_data), file_name) for file_name in file_names
]
zip_path, nodes_path, names_path, proteins_path, version_path = file_paths
```
 
Option 2 (your proposal made a little more concise):
```python
zip_path, nodes_path, names_path, proteins_path, version_path = [
    os.path.join(str(ncbi_data), file_name) for file_name in [
        "taxdmp.zip", "nodes.dmp", "names.dmp",
        "prot.accession2taxid.gz", "version.tsv"
    ]
]
```

Option 3 (original):
```python
zip_path = os.path.join(str(ncbi_data), "taxdmp.zip")
nodes_path = os.path.join(str(ncbi_data), "nodes.dmp")
names_path = os.path.join(str(ncbi_data), "names.dmp")
proteins_path = os.path.join(str(ncbi_data), "prot.accession2taxid.gz")
version_path = os.path.join(str(ncbi_data), "version.tsv")
```

I see how the 1st and 2nd options could be favored for their brevity. On the other hand, the 3rd option uses fewer lines and is more readable. I don't see a big gain in readability of brevity. I think I would choose to leave it as is. What do you think?

Reply:
That is very true. My suggestion makes it so much harder to read 😂
Leave it as is :)